fabric-combat-api-v1/src/main/java/net/fabricmc/fabric/mixin/combat/BowAttackMixin.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.combat;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.ai.RangedAttackMob;
+import net.minecraft.entity.mob.AbstractSkeletonEntity;
+import net.minecraft.entity.mob.IllusionerEntity;
+import net.minecraft.entity.mob.MobEntity;
+import net.minecraft.entity.projectile.PersistentProjectileEntity;
+import net.minecraft.entity.projectile.ProjectileUtil;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.combat.v1.ShotProjectileEvents;
+
+@Mixin({AbstractSkeletonEntity.class, IllusionerEntity.class})
+public abstract class BowAttackMixin extends MobEntity implements RangedAttackMob {
+	protected BowAttackMixin(EntityType<? extends MobEntity> entityType, World world) {
+		super(entityType, world);
+	}
+
+	@Redirect(method = "attack", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;spawnEntity(Lnet/minecraft/entity/Entity;)Z"))
+	public boolean modifyShotProjectile(World world, Entity persistentProjectileEntity, LivingEntity target, float pullProgress) {
```

Comment:
This will not replace the parameter, you need to use ModifyVariable in order to replace the old entity with a new one.

Reply:
doesnt the event system chain together the events then return the final result? here i am just redirecting the method call and then replacing the old entity with the events and then calling the spawn method with my new entity

Reply:
This is a mixin issue:

Here is an example of what happens with your mixin:

```java
void attack(params) {
    projectile = createProjectile();
    if (redirect$zzz000$modifyShotProjectile(projectile)) {
        ...
    }

    ...
}
```

The when you pass the parameter into the added redirect, you do not mutate the input parameter, if you change the parameter within your own mixin.

One way to consider this is every method is implicitly like this:
```java
public boolean modifyShotProjectile(final World world, final Entity persistentProjectileEntity, final LivingEntity target, final float pullProgress) {
```

You can set the value without it being final, but it acts more like a variable declared in your method which will drop out of scope when the mixin's method is exited.

Reply:
so if i just made a different local variable would that fix it?

Reply:
I probably did not eloquently describe this, but the simple fix is to change the projectile entity after this redirect by querying a sort of "temporary" field.